### ISSUE TITLE
Is pytorch xla spmd working as expected?

### ISSUE THREAD

**Issue by fengyang0317** [pytorch/xla#7846]
## 🐛 Bug
I tried to run [test_train_spmd_linear_model.py](https://github.com/pytorch/xla/blob/master/test/spmd/test_train_spmd_linear_model.py) with `sharding='batch'`. The input data sharing is {devices=[8,1]0,1,2,3,4,5,6,7}, which is expected. However, after a linear layer, the fc1 output sharding becomes 'replicated'. I am wonder whether all the following layers are running without sharding?

Print the sharding_spec during forward.
```
  def forward(self, x):
    print('x', torch_xla._XLAC._get_xla_sharding_spec(x))
    fc1 = self.fc1(x)
    print('fc1', torch_xla._XLAC._get_xla_sharding_spec(fc1))
    y = self.relu(fc1)
    print('y', torch_xla._XLAC._get_xla_sharding_spec(y))
    z = self.fc2(y)
    print('z', torch_xla._XLAC._get_xla_sharding_spec(z))
    o = self.fc3(z)
    print('o', torch_xla._XLAC._get_xla_sharding_spec(o))
    return o
```

Obtained outputs
```
x {devices=[8,1]0,1,2,3,4,5,6,7}
fc1 
y 
z 
o 
```

## To Reproduce
https://colab.research.google.com/drive/1508nWHxCthxWBlIeKLF0sLZXcjtsO6Ly#scrollTo=nGTxOOgDDOU3

run the colab above.

## Expected behavior

The fc1, y, z, o should have sharding.

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]: TPU
 - torch_xla version: nightly

**Comment by JackCaoG** [comment]
@alanwaketan can you take a look?

**Comment by alanwaketan** [comment]
This is expected. Sharding propagation happens during compilation time and not during tracing time. So if you do a mark_step, you should see the sharding annotation.

**Comment by fengyang0317** [comment]
Thank for your answer.

**Comment by fengyang0317** [comment]
@alanwaketan I just found that the GroupNorm, BatchNorm and etc. cannot keep the sharding. Is this expected?

```
import numpy as np
import torch
import torch_xla.amp.syncfree
import torch_xla.core.xla_model as xm
import torch_xla.distributed.spmd as xs
from torch_xla import runtime as xr

torch_xla.runtime.use_spmd()

device = xm.xla_device()
num_devices = xr.global_runtime_device_count()
device_ids = np.arange(num_devices)
mesh_shape = (num_devices, 1, 1, 1)
mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))

xt = torch.randn(128, 64, 64, 64).to(device)
xs.mark_sharding(xt, mesh, (0, 1, 2, 3))
gn_out = torch.nn.GroupNorm(32, 64).to(device)(xt)
conv_out = torch.nn.Conv2d(64, 64, 3, 1, 1).to(device)(xt)
xm.mark_step()
print(torch_xla._XLAC._get_xla_sharding_spec(xt))
print(torch_xla._XLAC._get_xla_sharding_spec(conv_out))  # Conv can keep sharding
print(torch_xla._XLAC._get_xla_sharding_spec(gn_out))

```

Output
```
{devices=[2,1,1,1]0,1}
{devices=[2,1,1,1]0,1}
{replicated}
```
